go/vt/mysqlctl/mysqlshellbackupengine.go

```diff
@@ -0,0 +1,340 @@
+	mysqlShellDumpFlags = `{"threads": 2}`
+	// flags passed to the Load command, as a JSON string
+	mysqlShellLoadFlags = `{"threads": 4, "updateGtidSet": "replace", "skipBinlog": true, "progressFile": ""}`
+	// drain a tablet when taking a backup
+	mysqlShellBackupShouldDrain = false
+	// disable redo logging and double write buffer
+	mysqlShellSpeedUpRestore = false
+
+	MySQLShellPreCheckError = errors.New("MySQLShellPreCheckError")
+)
+
+// MySQLShellBackupManifest represents a backup.
+type MySQLShellBackupManifest struct {
+	// BackupManifest is an anonymous embedding of the base manifest struct.
+	// Note that the manifest itself doesn't fill the Position field, as we have
+	// no way of fetching that information from mysqlsh at the moment.
+	BackupManifest
+
+	// Location of the backup directory
+	BackupLocation string
+	// Params are the parameters that backup was created with
+	Params string
+}
+
+func init() {
+	for _, cmd := range []string{"vtcombo", "vttablet", "vtbackup", "vttestserver", "vtctldclient"} {
+		servenv.OnParseFor(cmd, registerMysqlShellBackupEngineFlags)
+	}
+}
+
+func registerMysqlShellBackupEngineFlags(fs *pflag.FlagSet) {
+	fs.StringVar(&mysqlShellBackupLocation, "mysql_shell_backup_location", mysqlShellBackupLocation, "location where the backup will be stored")
+	fs.StringVar(&mysqlShellFlags, "mysql_shell_flags", mysqlShellFlags, "execution flags to pass to mysqlsh binary to be used during dump/load")
+	fs.StringVar(&mysqlShellDumpFlags, "mysql_shell_dump_flags", mysqlShellDumpFlags, "flags to pass to mysql shell dump utility. This should be a JSON string and will be saved in the MANIFEST")
+	fs.StringVar(&mysqlShellLoadFlags, "mysql_shell_load_flags", mysqlShellLoadFlags, "flags to pass to mysql shell load utility. This should be a JSON string")
+	fs.BoolVar(&mysqlShellBackupShouldDrain, "mysql_shell_should_drain", mysqlShellBackupShouldDrain, "decide if we should drain while taking a backup or continue to serving traffic")
+	fs.BoolVar(&mysqlShellSpeedUpRestore, "mysql_shell_speedup_restore", mysqlShellSpeedUpRestore, "speed up restore by disabling redo logging and double write buffer during the restore process")
+}
+
+// MySQLShellBackupEngine encapsulates the logic to implement the restoration
+// of a mysql-shell based backup.
+type MySQLShellBackupEngine struct {
+}
+
+const (
+	mysqlShellBackupBinaryName = "mysqlsh"
+	mysqlShellBackupEngineName = "mysqlshell"
+)
+
+func (be *MySQLShellBackupEngine) ExecuteBackup(ctx context.Context, params BackupParams, bh backupstorage.BackupHandle) (result BackupResult, finalErr error) {
+	params.Logger.Infof("Starting ExecuteBackup in %s", params.TabletAlias)
+
+	err := be.backupPreCheck()
+	if err != nil {
+		return BackupUnusable, vterrors.Wrap(err, "failed backup precheck")
+	}
+
+	start := time.Now().UTC()
+	location := path.Join(mysqlShellBackupLocation, params.Keyspace, params.Shard, start.Format("2006-01-02_15-04-05"))
```

Comment:
ah, simply because we were already using another alternative to taking logical backups outside of vitess and this follows the same format.

but to be fair, vitess already doesn't use a standard format either. backups are created using:
https://github.com/vitessio/vitess/blob/c51ee4b7e3da381843ca6ae37fa404715bf8d459/go/vt/mysqlctl/backup.go#L59-L60

I can switch to using that format if it makes more sense?

Reply:
Yes, let us use this format, and keep it consistent within the backups code.

Reply:
And I agree the format used by vitess is also non standard (and weird looking).

Reply:
this was changed to use the existing format!